### PR TITLE
Creator token revenue split tweaks

### PR DESCRIPTION
**Pull request by dobertRowneySr**
Specs: #3637 

#### Changes
- `MinRevenueSplitDuration` is now a storage variable
- `MinRevenueSplitForewarning` storage variable added. Now whenever `issue_revenue_split` is called (either with or without optional starting block) at `current_block` it holds `revenue_split_starting_block >= current_block + MinRevenueSplitForewarning`. 
- User participating in a previous split is able to unstake automatically by using `participate_in_split`

### CODE REVIEW

File: runtime-modules/project-token/src/lib.rs [hunk 1]
```diff
@@ -103,6 +100,11 @@ decl_storage! {
 
         /// Minimum duration of a token sale
         pub MinSaleDuration get(fn min_sale_duration) config(): T::BlockNumber;
+
+        /// Minimum revenue split duration constraint
+        pub MinRevenueSplitDuration get(fn min_revenue_split_duration) config(): T::BlockNumber;
+        /// Minimum revenue split forewarning constraint
+        pub MinRevenueSplitForewarning get(fn min_revenue_split_forewarning) config(): T::BlockNumber
```
Comment (Lezek123) [inline runtime-modules/project-token/src/lib.rs]: I think `TimeToStart` would be much clearer of a name than `Forewarning`

**Review by Lezek123 — CHANGES_REQUESTED** [review]
- Please add `issue_split_ok_with_custom_start_block` test case to test a scenario where the user provides `Some(start_block)`
- Not strictly related to this PR, but I think `issue_revenue_split` w/ `allocation == 0` should result in `DispatchError` (consistently w/ `participate_in_split` w/ `amount == 0`)
- Merge conflicts

File: runtime-modules/project-token/src/lib.rs (hunk 1 above)
Comment (dobertRowneySr) [reply runtime-modules/project-token/src/lib.rs]: fixed in 94d7272167f3cb56eebb85640dc8f66868a3ca28

**Review by Lezek123 — APPROVED** [review]
LGTM

**Pull request merged**